File: matiasmasso/Mat.NET/DT2/DTOApiLog.vb
Public Class DTOApiLog
    Public Enum Cods
        NotSet
        BritaxStoreLocator
    End Enum

    Public Property Cod As Cods
    Public Property Fch As Date
    Public Property Ip As String

    Shared Function Factory(oCod As DTOApiLog.Cods, Ip As String) As DTOApiLog
        Dim retval As New DTOApiLog
        With retval
            .Cod = oCod
            .Ip = Ip
        End With
        Return retval
    End Function
End Class

File: matiasmasso/Mat.NET/DTO/DTOApiLog.cs
using System;

namespace DTO
{
    public class DTOApiLog
    {
        public enum Cods
        {
            NotSet,
            BritaxStoreLocator
        }

        public Cods Cod { get; set; }
        public DateTime Fch { get; set; }
        public string Ip { get; set; }

        public static DTOApiLog Factory(DTOApiLog.Cods oCod, string Ip)
        {
            DTOApiLog retval = new DTOApiLog();
            {
                var withBlock = retval;
                withBlock.Cod = oCod;
                withBlock.Ip = Ip;
            }
            return retval;
        }
    }
}
